### DISCUSSION TITLE
Some random russian guy got Triton 2.0.0 to build on windows

### DISCUSSION THREAD

**Discussion by PleezDeez** [deepspeedai/DeepSpeed#2694] [General]
this gets us one step closer to a working windows version but it's still erroring out during install bc it's looking for lscpu info instead of Windows. I'm not totally inept at coding but my skills are subpar at best if anyone else feels like taking a crack at getting it installed. I know someone managed to get the build whl built but I have no idea where I saw it anymore... here's the triton 2.0.0 download. it didn't trigger any flags when scanning in Defender https://transfer.sh/me0xpC/triton-2.0.0-cp310-cp310-win_amd64.whl just download the file. activate whatever virtual environment and pip install the filename from your downloads folder

**Comment by 78Alpha** [comment]
`not found`

**Reply by jreoka** [reply]
Transfer.sh only retains files for 2 weeks. thats why

**Reply by jreoka** [reply]
try this link: https://huggingface.co/r4ziel/xformers_pre_built/resolve/main/triton-2.0.0-cp310-cp310-win_amd64.whl

**Reply by YuanFeis** [reply]
can i ues it in python3.9?

**Reply by jreoka** [reply]
@YuanFeis I honestly have no idea. Never tried it. You might have to recompile it, or it may just work.

**Reply by YuanFeis** [reply]
all right，thanks for your help

**Comment by cyberluke** [comment]
Hold my Pepsi...

you have only WHL :-D

triton-3.0.0-cp311-cp311-win_amd64.whl
triton-3.0.0-cp312-cp312-win_amd64.whl
triton-llvm-opt.exe
triton-lsp.exe
triton-opt.exe
triton-reduce.exe
triton.dll
zlib1.dll

Download here: https://1drv.ms/u/s!Aq5X0D4OWWATiKkx8gdY012BCa_wVQ?e=SaCr0l

What is so good on Triton anyway? They just rape vLLM. You should check out SGLang, very nice. I'm building my own OpenAI API gateway that will feature plug n play models and cover functionality of Ollama, LT Studio, OpenVINO and will feature nice dashboard with statistics and some Red Team System Prompts muhehehe.

**Comment by loadams** [comment]
Closing this discussion due to age and Windows whl having been published with version 0.15.0 on PyPI.

**Comment by cyberluke** [comment]
0.15.0 is half year old and for recent Pytorch and Microsoft and Intel AI toolkits, Cuda 12.4 and Python 3.12 build is required.

On top of that, that WHL:
- is not compiled properly
- don't have NVMe support even while Microsoft have DirectStorage SDK and it is supported in Windows 11 out of the box
- don't have AIO while Intel and Microsoft SDK have POSIX compatible implementation (apart from 2 API calls)
- it has some older Triton dependency and that also have some issues

You can close all other Windows discussions due to age as well - where you mostly promise Windows build.

This Deepseek is useless in current implementation. I will do it properly myself. Compile all Deepseek dependecies and then compile Deepseek with all the features.

I bought i9 Core Ultra 285K + RTX 4090 for $6000 and I even did not get to developing AI yet. Only preparing build system and libraries so something basic can run properly. The cost of the time I need to spend is getting high, so I might be a bit grumpy these days :)

**Reply by tjruwase** [reply]
@cyberluke, thanks for your comments. We greatly appreciate your frustration and candid feedback. We consider Windows critical and even though progress is slow, we are working on it. The main challenge to faster progress is limited bandwidth within the team. And even though we have already received a lot of support from the community, we need even more support to get Windows support on-par with Linux. Can you help us? We are very receptive to PRs, and DeepSpeed is a community project. 

> This Deepseek is useless in current implementation

On a different note, can you please edit the typos in your post from Deepseek to DeepSpeed? It would be unfair for Deepseek to share the blame of DeepSpeed :)

**Reply by loadams** [reply]
@cyberluke - I want to try and help with your feedback/comments above, but the 0.15.0 whl appears to be compiled fine for me.  I'm able to run a few test cases in DeepSpeedExamples.  Or are you referring to not having support built into that whl for `DS_BUILD_AIO`?

Could you also list, other than having newer whls built, what you'd like to see in them?  I believe the 0.15.0 whl is built with python 3.11, I know some users have requested 3.10, are you looking for older or newer python versions?  Same with Cuda versions?

**Comment by cyberluke** [comment]
Try to use Accelerate in Pytorch and select DeepSpeef using command:
accelerate config, it will crash due to AIO not present.

Right now I have 0.16.1 for Python 3.12


I dont want to waste your time, I am able to solve it myself and you know
the behavior at NVIDIA github is even worse because on website or in docs
they say cuFile.cu is in Cuda Toolkit, but it is only in Linux package, I
observe it for like 5 years and some people even wont get response to this
question. Our EU court cares more about citizens embracing social
capitalism. Google is here quite often.

Maybe someone will put NV to EU court. But I dont care, I only came here to
and to other companies to asses the situation. NV will be the biggest
problem, you and them could be represented in few Windows benchmarks that
you outperform competition and some Ph.d would also present positive
review, but after today I am not in that mood. I will just compile and pack
everything for Windows 11 community and then I wont care. Regarding missing
cuFile from NVIDIA, Copilot Pro said that you can load Linux .so shared
library in Windows Python using ctypes or cffi.

After this experience, I think the situation deserves small escalation and
more proactivity, so I think someone will reverse engineer or decompile
cuFile.so and make cuFile.dll - corporation arguments are just nonsense,
and either laziness or arrogancy. Copilot Pro said that it is ok to
decompile it, but it requires more skills. Im like Steve Jobs attitude, so
I will make it happen, it will take some time, but I will find someone with
good skill with Gihidra or IDA Pro.

In the end I will solve it for a lot complaining people on three american
github corporations and it will be a win win. Everyone gets peaceful life
and I can finally run some Pytorch fun, just spent $6000 on new Win AI pc
and only thing Im doing is solving someone else issues.

So please dont let yourself stress, customers will solve it themselves in
accordance to European Union law.
